engine/access/rpc/backend/backend_transactions.go

```diff
@@ -413,3 +419,32 @@ func (b *backendTransactions) getTransactionResultFromExecutionNode(
 func (b *backendTransactions) NotifyFinalizedBlockHeight(height uint64) {
 	b.retry.Retry(height)
 }
+
+func (b *backendTransactions) getTransactionResultFromAnyExeNode(ctx context.Context, execNodes flow.IdentityList, req execproto.GetTransactionResultRequest) (*execproto.GetTransactionResultResponse, error) {
+	var errors *multierror.Error
+	// try to execute the script on one of the execution nodes
+	for _, execNode := range execNodes {
+		resp, err := b.tryGetTransactionResult(ctx, execNode, req)
+		if err == nil {
+			return resp, nil
+		}
+		if status.Code(err) == codes.NotFound {
```

Comment:
so if the random exe node we chose returns not found, we don't bother looking at the others? I think that should work, though i think this might mean for most calls we only ask one exe node. but that's probably okay.

Reply:
I was thinking that getting an empty events from a EN who already has published an ER for the block containing the transaction should never happen hence I bail out with the first not found.